todo-list/src/hooks/todos/useTodo.ts

```diff
@@ -59,3 +70,60 @@ export function useAddTodo() {
     error: mutation.error,
   };
 }
+
+export function useUpdateTodo() {
+  const updatedTodoStore = useTodoStore((state) => state.updateTodo);
+  const mutation = useMutation<
+    Todo,
+    Error,
+    { todoId: number; title: string; completed: boolean }
+  >({
+    mutationFn: async ({ todoId, title, completed }) => {
+      const updatedTodo = await updateTodo(todoId, title, completed);
+      updatedTodoStore(updatedTodo);
+      return updatedTodo;
+    },
+    onError: (error) => {
+      console.error("Failed to update todo:", error.message);
+    },
+  });
+  const handleUpdateTodo = (
+    todoId: number,
+    title: string,
+    completed: boolean,
+  ) => {
+    mutation.mutate({ todoId, title, completed });
+  };

```

Comment:
Avoid using `.mutate()` when not providing `onSuccess, onSettled, onError`.

Use `.mutateAsync()` instead